applications/nrf5340_audio/src/audio/streamctrl.c

```diff
@@ -170,6 +170,24 @@ static void le_audio_rx_data_handler(uint8_t const *const p_data, size_t data_si
 	ret = data_fifo_pointer_first_vacant_get(&ble_fifo_rx, (void *)&iso_received, K_NO_WAIT);
 	ERR_CHK_MSG(ret, "Unable to get FIFO pointer");
 
+	uint32_t bitrate, sampling_rate, octets_per_frame;
+	static uint32_t data_size_mismatch_cnt;
+
+	if (CONFIG_AUDIO_DEV == HEADSET) {
+		ret = le_audio_config_get(&bitrate, &sampling_rate, &octets_per_frame);
+		if (ret) {
+			LOG_ERR("Failed to get configs");
+		}
+
+		if (data_size != octets_per_frame) {
```

Comment:
This should be checked directly in stream_recv_cb, no need to first get the octets per frame and then check when everything is already in the le_audio files

Reply:
Good point. 
That will significantly reduce the amount of changed code

Reply:
Pushed a new version that checks this in the le_audio files.